code/modules/mob/living/living.dm

```diff
@@ -224,18 +224,32 @@
 /mob/living/resist_grab(moving_resist)
 	if(!pulledby)
 		return
-	if(pulledby.grab_level)
-		if(prob(50))
-			playsound(src.loc, 'sound/weapons/thudswoosh.ogg', 25, 1, 7)
-			visible_message(SPAN_DANGER("[src] has broken free of [pulledby]'s grip!"), null, null, 5)
+	var/mob/user = src
+	// Passive grabs will also be broken, UNLESS pulledby has super strength, but this is ignored if you also have super strength
+	if(pulledby.grab_level || (!pulledby.grab_level && (HAS_TRAIT(pulledby, TRAIT_SUPER_STRONG) && !HAS_TRAIT(user, TRAIT_SUPER_STRONG))))
+		/// Chance for person to break free of grip, defaults to 50.
+		var/chance = 50
+		if(HAS_TRAIT(user, TRAIT_SUPER_STRONG) && !isxeno(pulledby)) // no extra chance to resist warrior grabs
+			chance += 30 // you are strong, you can overpower them easier
+		if(HAS_TRAIT(pulledby, TRAIT_SUPER_STRONG))
+			chance -= 30 // stronger grip
+		// above code means that if you are super strong, 80% chance to resist, otherwise, 20 percent. if both are super strong, standard 50.
+
+		if(prob(chance))
+			playsound(user.loc, 'sound/weapons/thudswoosh.ogg', 25, 1, 7)
+			visible_message(SPAN_DANGER("[user] has broken free of [pulledby]'s grip!"), null, null, 5)
 			pulledby.stop_pulling()
-			return 1
+			return TRUE
 		if(moving_resist && client) //we resisted by trying to move
-			visible_message(SPAN_DANGER("[src] struggles to break free of [pulledby]'s grip!"), null, null, 5)
-			client.next_movement = world.time + (10*pulledby.grab_level) + client.move_delay
+			visible_message(SPAN_DANGER("[user] struggles to break free of [pulledby]'s grip!"), null, null, 5)
+			// +1 delay if super strong, also done as passive grabs would have a modifier of 0 otherwise, causing spam
+			if(HAS_TRAIT(pulledby, TRAIT_SUPER_STRONG) && !HAS_TRAIT(user, TRAIT_SUPER_STRONG))
```

Comment:
Since `HAS_TRAIT(pulledby, TRAIT_SUPER_STRONG)` and `HAS_TRAIT(user, TRAIT_SUPER_STRONG)` are used multiple times in this proc, it would make sense to cache them as variables near the start. That way they only need to be called once rather than potentially three times.

For example:
```dm
	var/pulledby_is_strong = HAS_TRAIT(pulledby, TRAIT_SUPER_STRONG)
	var/src_is_strong = HAS_TRAIT(src, TRAIT_SUPER_STRONG)

	// Passive grabs will also be broken, UNLESS pulledby has super strength, but this is ignored if you also have super strength
	if(pulledby.grab_level || (pulledby_is_strong && !src_is_strong))
```